jdaviz/configs/default/plugins/virtual_observatory/vo_plugin.py

```diff
@@ -0,0 +1,419 @@
+        # If plugin is in "Manual" mode, we should never
+        # autocenter and potentially wipe the user's data
+        if self.viewer_selected == "Manual":
+            return
+
+        # If the user panned but tracking not enabled, don't recenter
+        # NOTE: float event typecheck assumes the only "float" type event triggering this method
+        #   is "zoom_center_x" and "zoom_center_y"
+        #   Otherwise, need some other way to determine the event = user panning
+        if isinstance(event, float) and not self.coord_follow_viewer_pan:
+            # Thus, we're no longer centered
+            self.viewer_centered = False
+            return
+
+        # gets the current viewer
+        if self.viewer_selected:
+            viewer = self.viewer.selected_obj
+
+        # nothing happens in the case there is no image in the viewer
+        # additionally if the data does not have WCS
+        if (
+            viewer.state.reference_data is None
+            or viewer.state.reference_data.coords is None
+        ):
+            self.source = ""
+            return
+
+        # Obtain center point of the current image and convert into sky coordinates
+        x_center = (viewer.state.x_min + viewer.state.x_max) * 0.5
+        y_center = (viewer.state.y_min + viewer.state.y_max) * 0.5
+        skycoord_center = viewer.state.reference_data.coords.pixel_to_world(
+            x_center, y_center
+        )
+
+        # Extract SkyCoord values as strings for plugin display
+        ra_deg = skycoord_center.ra.deg
+        dec_deg = skycoord_center.dec.deg
+        frame = skycoord_center.frame.name.lower()
+
+        # Show center value in plugin
+        self.source = f"{ra_deg} {dec_deg}"
+        self.coordframe_selected = frame
+
+        self.viewer_centered = True
+
+    @observe("waveband_selected", "change")
+    def vue_query_registry_resources(self, _=None):
+        """
+        Query Virtual Observatory registry for all SIA services
+        that serve data in that waveband around the source.
+        Then update the dropdown accordingly.
+        """
+        # If waveband was changed to nothing, immediately quit
+        # Don't throw an error due to trigger by plugin init
+        if not self.waveband_selected:
+            return
+
+        # Can't filter by coverage if we don't have a source to filter on
+        if self.resource_filter_coverage and not self.source:
+            error_msg = (
+                "Source is required for registry querying when coverage filtering is enabled. "
+                + (
+                    "Please enter your coordinates above "
+                    if self.viewer == "Manual"
+                    else f"Load data into viewer {self.viewer} first before querying "
+                )
+                + "or disable coverage filtering."
+            )
+            self.hub.broadcast(SnackbarMessage(error_msg, sender=self, color="error"))
+            raise ValueError(error_msg)
```

Comment:
what's the use for the snackbar and the exception?  This also seems to be in other cases - I can see the benefit for the developer, but is it useful for the user or could it interrupt the flow?  Without the spinner decorator (see comment below), these also should probably all reset any enabled "loading" booleans.

Reply:
Now that I have the spinner decorator, the main reason I have those exceptions are for the developers, your point exactly. If you'd like them removed, I can! But they're meant for if the user calls it from the API